test/suite/test_timestamp26.py

```diff
@@ -0,0 +1,550 @@
+        c[key] = ds.value(11)
+        self.session.commit_transaction()
+
+        key = ds.key(12)
+
+        # Insert a non-timestamped item, then update with a timestamp and then without a timestamp.
+        self.session.begin_transaction()
+        c[key] = ds.value(12)
+        self.session.commit_transaction()
+
+        self.session.begin_transaction()
+        self.session.timestamp_transaction('commit_timestamp=' + self.timestamp_str(2))
+        c[key] = ds.value(13)
+        self.session.commit_transaction()
+
+        self.session.begin_transaction()
+        c[key] = ds.value(14)
+        self.session.commit_transaction()
+
+        # Now alter the setting and make sure we detect incorrect usage. We must move the oldest
+        # timestamp forward in order to alter, otherwise alter will fail with EBUSY.
+        c.close()
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(10))
+        config = 'assert=(write_timestamp=on)'
+        self.session.alter(uri, 'write_timestamp_usage=ordered,' + config)
+
+        c = self.session.open_cursor(uri)
+        key = ds.key(15)
+
+        # Detect decreasing timestamp.
+        self.session.begin_transaction()
+        c[key] = ds.value(15)
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(15))
+
+        msg = 'with an older timestamp'
+        self.session.begin_transaction()
+        self.session.timestamp_transaction('commit_timestamp=' + self.timestamp_str(14))
+        c[key] = ds.value(16)
+        with self.expectedStderrPattern(msg):
+            self.session.commit_transaction()
+
+        # Detect not using a timestamp.
+        msg = 'use timestamps once they are first used'
+        self.session.begin_transaction()
+        c[key] = ds.value(17)
+        with self.expectedStderrPattern(msg):
+            self.session.commit_transaction()
+
+        # Now alter the setting again and detection is off.
+        c.close()
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(20))
+        self.session.alter(uri, 'assert=(write_timestamp=off)')
+        c = self.session.open_cursor(uri)
+        key = ds.key(18)
+
+        # Detection is off we can successfully change the same key with then without a timestamp.
+        self.session.begin_transaction()
+        c[key] = ds.value(18)
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(21))
+
+        self.session.begin_transaction()
+        c[key] = ds.value(19)
+        self.session.commit_transaction()
+        c.close()
+
+# Test timestamp settings with inconsistent updates.
+class test_timestamp26_ts_inconsistent(wttest.WiredTigerTestCase):
+    types = [
+        ('fix', dict(key_format='r', value_format='8t')),
+        ('row', dict(key_format='S', value_format='S')),
+        ('var', dict(key_format='r', value_format='S')),
+    ]
+    scenarios = make_scenarios(types)
+
+    def XXXtest_timestamp_inconsistent(self):
```

Comment:
What is the point of leaving this in here? Is it supposed to be made real before merge and was forgotten?

Reply:
Nice catch! I missed one of the tests when I was updating them. Fixed in 6487e0adb.